src/releaseNotesProvider.ts

```diff
@@ -43,16 +43,16 @@ async function showReleaseNotesImpl(
 ): Promise<Either<string, void>> {
   const state = context.globalState;
 
-  const remote = isRemote();
-  if (remote.kind === "left") {
-    return remote;
-  }
-
-  const version = getVersion(calledOn);
+  const version = getVersion();
   if (version.kind === "left") {
     return version;
   }
 
+  const remote = isRemote();
+  if (calledOn === "onExtensionStart" && remote.kind === "left") {
```

Comment:
```suggestion
  if ((calledOn === "onExtensionStart" && remote.kind === "left") || calledOn === "onUserDemand") {
```

I prefer to check `calledOn === "onUserDemand"` outside of `isRemote` function, because `"onUserDemand"` is irrelevant to the remote environment.